middle_end/flambda/compare/compare.ml

```diff
@@ -766,6 +766,25 @@ let function_decls env decl1 decl2 : unit Comparison.t =
   else Different { approximant = () }
 ;;
 
+(** Match up equal elements in two lists and iterate through both of them,
+    using [f] analogously to [Map.S.merge] *)
+let iter2_merged l1 l2 ~compare ~f =
+  let l1 = List.sort compare l1 in
+  let l2 = List.sort compare l2 in
+  let rec go l1 l2 =
+    match l1, l2 with
+    | []      , []       -> ()
```

Comment:
Nitpick: I find the alignment of the different blocks hard to read -- at some point I thought the `; go ....` part was a comment.

Reply:
Yeah, looking at it now, it's not helping. I've moved everything to separate lines and taken out the alignment.